azurelinuxagent/common/osutil/default.py

```diff
@@ -1191,6 +1193,67 @@ def get_mount_point(self, mountlist, device):
                     return tokens[2] if len(tokens) > 2 else None
         return None
 
+    @staticmethod
+    def _enumerate_device_id():
+        """
+        Enumerate all storage device IDs.
+
+        Args:
+            None
+
+        Returns:
+            Iterator[Tuple[str, str]]: VmBus and storage devices.
+        """
+
+        if os.path.exists(STORAGE_DEVICE_PATH):
+            for vmbus in os.listdir(STORAGE_DEVICE_PATH):
+                deviceid = fileutil.read_file(os.path.join(STORAGE_DEVICE_PATH, vmbus, "device_id"))
+                guid = deviceid.strip('{}\n')
+                yield vmbus, guid
+
+    @staticmethod
+    def search_for_resource_disk(gen1_device_prefix, gen2_device_id):
```

Comment:
>def search_for_resource_disk(gen1_device_prefix, gen2_device_id): [](start = 4, length = 65)

Is there an easy way to detect if this is a Gen2VM, if so would it make more sense to refactor this into a flow where we detect whether this is a Gen2 or Gen1, if it's Gen2 there is a different search_for_resource_disk function (sort of like function pointers, one for Gen1, one for Gen2).

Reply:
Good question, I'll see if I can find that answer.

Reply:
Not sure you still need it, just for your reference.
 if [ -d /sys/firmware/efi/ ]; then
        os_GENERATION=2
    else
        os_GENERATION=1
    fi